### PR TITLE
Updates for Mongoose

### PR DESCRIPTION
**Pull request by simon-xiong-ca**
1.  Definition of mongoose.Model uses interfaces again instead of type intersection since type intersections were hard to read and could not be extended by another interface.
2. Updated to mongoose 4.5.9
3. Added missing properties on mongoose.Model
4. Was some confusion about how to use the definitions so I added a README.
5. Updated mongoose.Promise to use global.Promise by default, and allowing users to manually override if necessary.
6. Updated related mongoose definitions: passport-local-mongoose, mongoose-paginate, mongoose-promise.

### CODE REVIEW

**Comment by simonxca** [comment]
@horiuchi

File: mongoose/README.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+});
+
+interface IUser extends Document {
+  username: string;
+  age: number;
+  friends: string[];
+  data: any[];
+}
+
+var UserModel: Model<IUser> = mongoose.model<IUser>('User', UserSchema);
```
Comment (horiuchi) [inline mongoose/README.md]: Please check the compile result.
I thought the `mongoose.` is not required.

**Comment by horiuchi** [comment]
LGTM 👍

**Comment by horiuchi** [comment]
@simonxca Thanks!

**Pull request merged**